### PR TITLE
Allow discovery of non-ready Pods

### LINKED ISSUE #96: Allow non-ready pods to be sync'd
Currently, the agent ignores (and in turn removes) pod entities in a non-ready state. The non-ready state can be useful in monitoring.

### PR DESCRIPTION
**Pull request by mohabusama**
Allow discovery of non-ready Pods

Resolve #96

### CODE REVIEW

**Comment by mohabusama** [comment]
@aermakov-zalando Please take a look

**Comment by aermakov-zalando** [comment]
I'm not sure if `conditions` will be present in a completely new pod. Might be safer to use `get('conditions', [])`.

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -289,13 +286,11 @@ def get_cluster_pods_and_containers(
...
     for pod in pods:
-        if not pod.ready:
-            continue
 
         obj = pod.obj
 
         containers = obj['spec'].get('containers', [])
-        container_statuses = {c['name']: c for c in obj['status']['containerStatuses']}
+        container_statuses = {c['name']: c for c in obj['status'].get('containerStatuses', {})}
         conditions = {c['type']: c['status'] for c in obj['status']['conditions']}
```
Comment (aermakov-zalando) [inline zmon_agent/discovery/kubernetes/cluster.py]: I'm not sure if `conditions` will be present in a completely new pod. Might be safer to use `get('conditions', [])`.

Comment (mohabusama) [reply zmon_agent/discovery/kubernetes/cluster.py]: My tests showed they exist. But I think it makes sense to be safe here.

**Comment by mohabusama** [comment]
@vetinari please merge this when ready.

**Review by vetinari — APPROVED** [review]

**Comment by vetinari** [comment]
:+1:

**Comment by pitr** [comment]
👍

**Pull request merged**